e2e/pollers.go

```diff
@@ -52,7 +52,7 @@ func pollExecOnVM(ctx context.Context, kube *kubeclient, vmPrivateIP, jumpboxPod
 	})
 
 	if err != nil {
-		return nil, err
+		return execResult, err
```

Comment:
or in the method?

Reply:
yeah, I'd personally rather keep the explicit nil here and add the check to the method, what do you think @UtheMan

Reply:
We could do that, but I feel like it would clutter the method?

```go
		execResult, err := pollExecOnVM(ctx, opts.kube, privateIP, podName, sshPrivateKey, command)
		if err != nil {
                        if execResult != nil {
			execResult.dumpAll()
                        }
			return fmt.Errorf("unable to execute validator command %q: %s", command, err)
		}

		if execResult != nil &&  execResult.exitCode != "0" {
			execResult.dumpAll()
			return fmt.Errorf("validator command %q terminated with exit code %s", command, execResult.exitCode)
		}
```

would have to be something like this I think?

Reply:
depends on whether we think there could be instances where execResult is nil while err is also nil

I was thinking we'd do something like

if r != nil {
  r.dumpStdout()
  r.dumpStderr()
}

in r.dumpAll(), while keeping the other stuff as is